### PR TITLE
improve(GithubAuth): Add authenticated to app record, rework auth flow.

### PR DESCRIPTION
**Pull request by jdetle**
This is initial work towards fixing the github authorization flow. I think adding 'authenticated' to the app record is a good idea. Initially my thought was to go with a function in src/main/menu called publishAuth:

``` javascript
export function githubAuth(item, focusedWindow) {
  const win = new BrowserWindow({ show: false,
                                  webPreferences: { zoomFactor: 0.75 } });
  win.webContents.on('dom-ready', () => {
    if (win.getURL().indexOf('callback?code=') !== -1) {
      win.webContents.executeJavaScript(`
        require('electron').ipcRenderer.send('auth', document.body.textContent);
        `);
      ipc.on('auth', (event, auth) => {
        const authorization = JSON.parse(auth);
        send(focusedWindow, 'menu:publish:auth', authorization.access_token);
        win.close();
        return;
      });
    } else {
      win.show();
    }
  });
  win.loadURL('https://nteract-oauth.now.sh/github');
}
```

and:

``` javascript
publishAuthAndGist(){
   publishAuth();
   createSender('menu:publish:gist');
}
```

but theres some issue with synchronicity. Gonna pair with @captainsafia on this right now.

### CODE REVIEW

File: src/notebook/epics/github-publish.js [hunk 1]
```diff
@@ -105,7 +106,17 @@ export function publishNotebookObservable(github, notebook, filepath, notificati
...
     const files = {};
     files[filename] = { content: notebookString };
+    if (authenticated) {
+      const email = github.users.get({}, function(err, res) {
+        console.log(err, res);
+      });
 
+      notificationSystem.addNotification({
+        title: 'Authenticated',
+        message: 'Authenticated as `${ email}`',
```
Comment (rgbkrk) [inline src/notebook/epics/github-publish.js]: I'd use their username instead of their email address.

**Comment by jdetle** [comment]
publish-that-gist

**Comment by jdetle** [comment]
Oops, inadvertently created some modern art.

**Comment by jdetle** [comment]
publish-gist

**Comment by captainsafia** [comment]
Now that I think about it we probably want to use a different term from `Authenticated Gist`. I'm not sure what the best word would be but maybe `User Gist` or something?

**Comment by rgbkrk** [comment]
Slick!

**Comment by jdetle** [comment]
```
Publish 
> Gist Anonymously 
> Gist to Github Profile
```

**Comment by rgbkrk** [comment]
User Gist makes sense.

**Comment by captainsafia** [comment]
You need to update your test cases as well.

**Pull request merged**